Robust.Client/GameObjects/Components/Renderable/SpriteComponent.cs

```diff
@@ -1622,14 +1623,13 @@ public string GetDebugString()
         }
 
         /// <inheritdoc/>
-        public Box2 CalculateBoundingBox(Matrix3 worldMatrix)
+        public Box2Rotated CalculateRotatedBoundingBox(Vector2 worldPosition, Angle worldRotation, IEye? eye = null)
         {
-            // Given if the thing is upside down the offset is inverse we need to do a matrix transform
-            var worldPos = new Vector2(worldMatrix.R0C2, worldMatrix.R1C2);
-
             // fast check for empty sprites
-            if (Layers.Count == 0)
-                return new Box2(worldPos, worldPos);
+            if (!Visible || Layers.Count == 0)
+            {
+                return new Box2Rotated(new Box2(worldPosition, worldPosition));
```

Comment:
With Box2Rotated you'll also want to set the Origin because that's what all the points get rotated around (which defaults to Vector2.Zero).

Reply:
Given the rotation angle is always zero here, I'm not sure if the rotation origin actually matters, but I've added it anyways.